tests/Http/HttpRequestTest.php

```diff
@@ -537,4 +537,22 @@ public function testCreateFromBase()
 
         $this->assertEquals($request->request->all(), $body);
     }
+
+    public function testJsonContent()
+    {
+        $body = [
+            'foo' => 'bar',
+            'baz' => ['qux'],
+        ];
+
+        $server = [
+            'CONTENT_TYPE' => 'application/json',
```

Comment:
The ServerBag does the translation too:

https://github.com/symfony/HttpFoundation/blob/2.7/ServerBag.php#L31